binding/Binding/SKManagedStream.cs

```diff
@@ -90,18 +87,25 @@ protected override void DisposeManaged ()
 
 		private IntPtr OnReadManagedStream (IntPtr buffer, IntPtr size)
 		{
-			byte[] managedBuffer;
-			using (var reader = new BinaryReader (stream, Encoding.UTF8, true)) {
-				managedBuffer = reader.ReadBytes ((int)size);
-			}
-			var result = managedBuffer.Length;
-			if (buffer != IntPtr.Zero) {
-				Marshal.Copy (managedBuffer, 0, buffer, result);
-			}
-			if (!stream.CanSeek && (int)size > 0 && result <= (int)size) {
+			if (buffer == IntPtr.Zero)
+				throw new ArgumentNullException (nameof (buffer));
+			if ((int)size < 0)
+				throw new ArgumentOutOfRangeException (nameof (size));
+
+			if (size == IntPtr.Zero)
+				return IntPtr.Zero;
+
+			using var managedBuffer = Utils.RentArray<byte> ((int)size);
+			var len = stream.Read (managedBuffer.Array, 0, managedBuffer.Length);
```

Comment:
Shouldn't this be `var len = stream.Read (managedBuffer.Array, 0, size)`;?

The `managedBuffer.Length` is always the first power of two larger than `size` I believe

Reply:
That would be the case if I was using `ArrayPool` directly. I am using my Utils version that makes sure to set the `Length` of the "wrapper" to what I requested.

Reply:
Thanks for the review!

Reply:
Oh right! I actually remember you making this utils method, I forgot about that.